sharktank/sharktank/examples/export_paged_llm_v1.py

```diff
@@ -61,7 +62,11 @@ def main():
     llama_config = LlamaModelConfig(hp)
     llama_config.use_hf = False
     llama_config.static_tables = False  # Rely on the compiler for hoisting tables.
-    llama_config.kv_cache_type = "direct" if args.bs == [1] else "paged"
+    llama_config.kv_cache_type = "paged"
+
+    attn_q_weight = dataset.root_theta.tensor("blk")["0"]["attn_q"]["weight"]
+    if isinstance(attn_q_weight, SplitPrimitiveTensor):
+        llama_config.tensor_parallelism_size = attn_q_weight.shard_count
```

Comment:
This looks a bit brittle. Can dataset.properties carry the tensor_parralelism_size? If it is too much trouble I guess we can live with it.

Reply:
So the problem with adding it to `data.properties`  is that its not a default part of `gguf` and we are already plumbing it via the `LLamaHParams`. I attempted to wire it further down but it just turned into spaghetti coding.

Long term we want to load from the same `irpa` file so we will want to re-evaluate this for now.